rasa/core/policies/ted_policy.py

```diff
@@ -609,10 +630,14 @@ def predict_action_probabilities(
         confidence, is_e2e_prediction = self._pick_confidence(confidences, similarities)
 
         if self.config[LOSS_TYPE] == SOFTMAX and self.config[RANKING_LENGTH] > 0:
-            confidence = rasa.utils.train_utils.normalize(
+            confidence = rasa.utils.train_utils.sort_and_rank(
                 confidence, self.config[RANKING_LENGTH]
             )
 
+            if self.config[RELATIVE_CONFIDENCE]:
```

Comment:
`cosine` similarity should not be normalized?

Reply:
Good point! I added a check for this but I also added a warning if the user has configured to use `cosine` with `cross_entropy` loss or `inner` with `margin` loss.